modules/images/dominant-color/load.php

```diff
@@ -0,0 +1,425 @@
+<?php
+/**
+ * Module Name: Dominant Color
+ * Description: Adds support to store dominant-color for an image and create a placeholder background for that color.
+ * Experimental: Yes
+ *
+ * @package performance-lab
+ * @since 1.0.0
+ */
+
+/**
+ * Add Dominant Color preload support to WordPress.
+ *
+ * TODO: Add tests
+ */
+class wp_Dominant_Color {
+
+
+	public function __construct() {
+
+		add_filter( 'wp_print_scripts', [ $this, 'add_styles' ] );
+		add_filter( 'wp_footer', [ $this, 'add_scripts' ] );
+
+		add_filter( 'wp_generate_attachment_metadata', [ $this, 'dominant_color_metadata' ], 10, 2 );
+		add_filter( 'wp_generate_attachment_metadata', [ $this, 'has_transparency_metadata' ], 10, 2 );
+
+		// do we have the new filter or are duplicating core the functions?
+		if ( has_filter( 'wp_img_tag_add_adjust' ) ) {
```

Comment:
Has_filter does not work this way. It check if something has hooked into that filter. Let's remove this until that PR is merged into core.